### PR TITLE
(docs): fix typos

### PR DESCRIPTION
**Pull request by kianmeng**

### CODE REVIEW

**Review by HJianBo — APPROVED** [review]
@kianmeng  Thanks for your detailed efforts to fix so many spelling errors.

It's ok for me

**Comment by zmstone** [comment]
@kianmeng thank you so much.
Please change the commit message to:
`docs: fix typo`

File: apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl [hunk 1]
```diff
@@ -740,7 +740,7 @@ t_sqlparse_foreach_2(_Config) ->
 t_sqlparse_foreach_3(_Config) ->
     %% Verify foreach-incase with and without 'AS'
     Sql = "foreach payload.sensors as s "
-          "incase s.cmd != 1 "
+          "in case s.cmd != 1 "
```
Comment (k32) [inline apps/emqx_rule_engine/test/emqx_rule_engine_SUITE.erl:743]: This doesn't look like a valid change

**Review by k32 — CHANGES_REQUESTED** [review]
Thanks for the PR. Please revert changes in the SQL queries.

File: apps/emqx_gateway/src/coap/emqx_coap_tm.erl [hunk 2]
```diff
@@ -66,7 +66,7 @@
           outgoing => coap_message(),
           timeouts => list(ttimeout()),
           has_sub  => undefined | sub_register(),
-          transport => emqx_coap_transport:transprot()}.
+          transport => emqx_coap_transport:transport()}.
```
Comment (zmstone) [inline apps/emqx_gateway/src/coap/emqx_coap_tm.erl:69]: @HJianBo I wonder why this was not found by dialyzer

**Comment by zmstone** [comment]
I did a force push to @kianmeng's fork.

**Review by terry-xiaoyu — APPROVED** [review]

**Review by k32 — APPROVED** [review]

**Comment by kianmeng** [comment]
:tada: :tada: :tada: :tada: :tada:

**Pull request merged**